bin/p2-launch/main.go

```diff
@@ -87,9 +93,27 @@ func main() {
 	fetcher := uri.BasicFetcher{Client: httpClient}
 
 	podFactory := pods.NewFactory(*podRoot, types.NodeName(*nodeName), fetcher, *requireFile, pods.NewReadOnlyPolicy(false, nil, nil))
+	dockerClient, err := dockerclient.NewEnvClient()
+	if err != nil {
+		log.Fatalf("could not create docker client: %s", err)
+	}
+	podFactory.SetDockerClient(*dockerClient)
 	pod := podFactory.NewLegacyPod(manifest.ID())
 
-	err = pod.Install(manifest, auth.NopVerifier(), artifact.NewRegistry(*artifactRegistryURL, fetcher, osversion.DefaultDetector))
+	containerRegistryAuthStr := ""
+	if *containerRegistryJsonKeyFile != "" {
+		data, err := ioutil.ReadFile(*containerRegistryJsonKeyFile)
+		if err != nil {
+			log.Fatalf("could not read container json key file %s: %s", *containerRegistryJsonKeyFile, err)
+		}
+		authConfig := dockertypes.AuthConfig{
+			Username: "_json_key",
```

Comment:
is this a common thing all docker users will have? also should this be a constant in pkg/docker?